packages/rum-core/src/browser/locationChangeObservable.ts

```diff
@@ -1,23 +1,34 @@
-import { monitor, addEventListener, DOM_EVENT } from '@datadog/browser-core'
+import { monitor, addEventListener, DOM_EVENT, Observable } from '@datadog/browser-core'
 
-export function trackLocationChanges(onLocationChange: () => void) {
-  const { stop: stopHistoryTracking } = trackHistory(onLocationChange)
-  const { stop: stopHashTracking } = trackHash(onLocationChange)
+export interface LocationChange {
+  oldLocation: Readonly<Location>
+  newLocation: Readonly<Location>
+}
 
-  return {
-    stop: () => {
+export function createLocationChangeObservable(location: Location) {
+  let currentLocation = { ...location }
+  const observable = new Observable<LocationChange>(() => {
+    const { stop: stopHistoryTracking } = trackHistory(onLocationChange)
+    const { stop: stopHashTracking } = trackHash(onLocationChange)
+    return () => {
       stopHistoryTracking()
       stopHashTracking()
-    },
+    }
+  })
+
+  function onLocationChange() {
```

Comment:
Nit: What do you think of moving this function inside the observable to slowly move toward an implementation where `notify()` would not be exposed?

Reply:
FMU, it does not change anything for now. If we choose to go this way at some point, it would only be one easy change more to do, not a big deal IMO.

Reply:
Fair enough!